calculating weighted diameter

wintun24 opened the pull request:
Hi @alysclark 

Here I did the pull request for adding

1. the diameter calculation of tree branch (pre-capillary)
2. the diameter calculation of conduits (follow your concept of conduit volume)
3. weighted diameter for whole tree 
4. Unittesting for all of them

Thanks.

wintun24 committed:
calculating weighted diameter

alysclark reviewed (changes requested):
Hi Win, Please add or modify the documentation for the routine or routines you have changed, in the format outlined here: http://placentagen.readthedocs.io/en/latest/
There are a lot of whitespace changes in this and I am not sure why they are added. It is better to do multiple commits with fewer changes, especially when changing formating so the reviewer of the changes can see 'real changes to the code' vs format changes.

wintun24 commented:
@alysclark .. Ok, I will work on documentation. 

Regarding the whitespace changes, that happens because I added "print" lines here and there to confirm what the outputs are, and then when I commit back, I deleted those "print" lines. That's the reason for many whitespace adding here and there. 

In this current code, please ignore those white space. In the future, I will be careful not to change formatting if not required. 

Thanks.

wintun24 committed:
adding documentation of subroutines
